torchtitan/config_manager.py

```diff
@@ -207,6 +207,62 @@ def __init__(self):
             default=1,
             help="Pipeline Parallelism degree. 1 means disabled.",
         )
+        self.parser.add_argument(
+            "--experimental.pipeline_parallel_stages_per_rank",
+            type=int,
+            default=1,
+            help="""
+                Pipeline Parallelism number of stages per rank (a.k.a. virtual stages)
+
+                For simple schedules, this should be 1.
+
+                For looped schedules, this can be greater than one.
+
+                If the number of stages produced by splitting does not match the expected number of stages,
+                an error will be raised for sanity.""",
+        )
+        self.parser.add_argument(
+            "--experimental.pipeline_parallel_split_points",
```

Comment:
Am I right that in some vague sense: out of the three configs `training.pipeline_parallel_degree`, `experimental.pipeline_parallel_stages_per_rank`, and `experimental.pipeline_parallel_split_points`, given any two, the remaining one can be inferred?

As a user I may not care too much about where the exact split points are (as long as it's roughly even), so can we just determine the split points given pp-degree and number of stages per rank.

Reply:
I think to keep things simple, since we did not yet support virtual stages or looped schedules in torchtitan, this argument (num_stages) can be deleted for now.

In the future, I think we don't even need to add this num_stages arg unless we support automated splitting.  If the user gives split points (as is required now), we can require the number of split points be divisible by the number of ranks for looped schedules or equal the number of ranks for regular schedules.